### PR TITLE
Fix Babel config to use correct browserslist config

### PR DESCRIPTION
**Pull request by swissspidy**
While looking at #859 I realized that we're using `@wordpress/babel-preset-default` which loads the WP browserslist config instead of our own. It's more extensive, so not per se an issue.

### CODE REVIEW

File: assets/src/edit-story/app/media/mediaProvider.js [hunk 1]
```diff
@@ -53,6 +53,9 @@ function MediaProvider({ children }) {
     removeProcessing,
     updateMediaElement,
   } = actions;
+  const {
```
Comment (spacedmonkey) [inline assets/src/edit-story/app/media/mediaProvider.js]: Is this related to fix?

Comment (swissspidy) [reply assets/src/edit-story/app/media/mediaProvider.js]: Sort of. Tests failed otherwise because a function was used before initialization. Not sure why it works right now.

Comment (swissspidy) [reply assets/src/edit-story/app/media/mediaProvider.js]: Actually, unrelated. Storybook is currently broken: https://google.github.io/web-stories-wp/?path=/story/playground-stories-editor--default

Will fix separately.

Comment (swissspidy) [reply assets/src/edit-story/app/media/mediaProvider.js]: Hmm looks like it was already fixed though. But the storybook wasn't deployed.

File: babel.config.js [hunk 2]
```diff
@@ -14,30 +14,46 @@
...
 
-/**
- * WordPress dependencies
- */
-const defaultConfig = require('@wordpress/babel-preset-default');
-
 module.exports = function (api) {
-  const config = defaultConfig(api);
+  const isTest = api.env('test');
+  const isProduction = api.env() === 'production';
```
Comment (miina) [inline babel.config.js]: Should this be `api.env('production')` for consistency?

Comment (swissspidy) [reply babel.config.js]: Makes sense. Updated now.

**Review by miina — APPROVED** [review]
LGTM
